server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java

```diff
@@ -53,10 +53,15 @@ public void testCatalogCRUD() throws ApiException {
         CatalogInfo catalogInfo2 = catalogOperations.getCatalog(CATALOG_NAME);
         Assert.assertEquals(catalogInfo, catalogInfo2);
 
-        // Update catalog
-        System.out.println("Testing update catalog..");
-        CatalogInfo updatedCatalogInfo = catalogOperations.updateCatalog(CATALOG_NAME, CATALOG_NEW_NAME, CATALOG_NEW_COMMENT);
-        assertCatalog(updatedCatalogInfo, CATALOG_NEW_NAME, CATALOG_NEW_COMMENT);
+        // Update catalog name
+        System.out.println("Testing update catalog: changing name..");
+        CatalogInfo updatedCatalogInfo = catalogOperations.updateCatalog(CATALOG_NAME, CATALOG_NEW_NAME, null);
+        assertCatalog(updatedCatalogInfo, CATALOG_NEW_NAME, COMMENT);
+
+        // Update catalog comment
+        System.out.println("Testing update catalog: changing comment..");
+        CatalogInfo updatedCatalogInfo2 = catalogOperations.updateCatalog(CATALOG_NEW_NAME, null, CATALOG_NEW_COMMENT);
+        assertCatalog(updatedCatalogInfo2, CATALOG_NEW_NAME, CATALOG_NEW_COMMENT);
```

Comment:
Perhaps we can add another test case that an exception is thrown when nothing is provided to be updated?

Reply:
I added a test for an exception being thrown now and changed [`CatalogRepository.updateCatalog`](https://github.com/unitycatalog/unitycatalog/blob/26657122db5e0363e4988ef9c126c541848eeb63/server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java#L102-L104) to also throw an exception if there is nothing to update.

The one thing I'm not quite happy with in my fix is that I had to change the exception that the CLI `updateCatalog` handler throws to be an `ApiException` instead of `CliException` [here](https://github.com/unitycatalog/unitycatalog/blob/26657122db5e0363e4988ef9c126c541848eeb63/examples/cli/src/main/java/io/unitycatalog/cli/CatalogCli.java#L82). This is because a `CliException` gets handled above so the test for an exception being thrown would not pass with a `CliException` 😅 This feels a bit wrong to me, but it works.

Actually, [all of this](https://github.com/unitycatalog/unitycatalog/blob/26657122db5e0363e4988ef9c126c541848eeb63/examples/cli/src/main/java/io/unitycatalog/cli/CatalogCli.java#L74-L82) is not necessarily needed now that the server itself checks if there is anything to update. I still left this in so that we can print a much more useful error message.

What do you think, is this fine or is there a cleaner way?

Reply:
Agree with everything you said. All looks good, but yeah a small nit would be the CliException, I think we should change the logic [here](https://github.com/unitycatalog/unitycatalog/blob/main/examples/cli/src/main/java/io/unitycatalog/cli/UnityCatalogCli.java#L94) and let CliException be handled the same as ApiException. Because as of now the CliException just does a print of the exception and that turns out to not be as descriptive.

Reply:
Handling a CliException the same as an ApiException sounds very good to me; added this change and changed the throwing from ApiException to CliException.